mapreduce/src/main/java/uk/gov/gchq/palisade/PalisadeInputSplit.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2018 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.util.StringUtils;
+
+import uk.gov.gchq.palisade.jsonserialisation.JSONSerialiser;
+import uk.gov.gchq.palisade.resource.Resource;
+import uk.gov.gchq.palisade.service.request.ConnectionDetail;
+import uk.gov.gchq.palisade.service.request.DataRequestResponse;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * The input split for {@link PalisadeInputFormat}. This class contains all the information for describing the resources
+ * for one split and the necessary connection methods for finding the data services responsible for finding those
+ * resources.
+ */
+public class PalisadeInputSplit extends InputSplit implements Writable {
```

Comment:
I think you need a setter for requestResponse or add JsonCreator to main constructor otherwise Jackson won't be able to set the requestResponse field. Please could you also add a simple test for this class where it serialises and deserialises the class.

Reply:
Done